### PR TITLE
Fix a crash when inferring a `typing.TypeVar` call.

### PR DESCRIPTION
**Pull request by mbyrnepr2**
Closes pylint-dev/pylint#8802

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
|    | :sparkles: New feature |
| ✓   | :hammer: Refactoring   |
|    | :scroll: Docs          |

## Description

Closes https://github.com/pylint-dev/pylint/issues/8802

### CODE REVIEW

File: tests/test_typing.py
Comment (mbyrnepr2) [inline tests/test_typing.py:1]: Created this module since it felt like all brain tests were going into brain/test_brain.py - let me know what you think.

File: tests/brain/test_typing.py
Comment (mbyrnepr2) [inline tests/brain/test_typing.py:1]: Created this module since it felt like all brain tests were going into brain/test_brain.py - let me know what you think.

**Comment by jacobtylerwalls** [comment]
Let's backport. Shall we then move the changelog to 2.15.6?

File: astroid/brain/brain_typing.py [hunk 1]
```diff
@@ -134,7 +134,14 @@ def infer_typing_typevar_or_newtype(node, context_itton=None):
         raise UseInferenceDefault
 
     typename = node.args[0].as_string().strip("'")
-    node = extract_node(TYPING_TYPE_TEMPLATE.format(typename))
+    node = ClassDef(
```
[the remark this answers is not in the document]
Comment (mbyrnepr2) [reply astroid/brain/brain_typing.py:137]: Thanks for such quick feedback👍.
I started a short vacation right after opening this so I’ll think about a test case when I can.

[the remark this answers is not in the document]
Comment (mbyrnepr2) [reply astroid/brain/brain_typing.py:137]: Hey @DanielNoord I've taken a look finally :)

It seems to me there is no need for `__getitem__` to be attached to the astroid node representing either `typing.TypeVar` or `typing.NewType` since neither of these are subscriptable. What do you think?
Similar [point](https://github.com/python/cpython/blob/3.11/Lib/typing.py#L1323C28-L1323C28) for `__args__` since that is used to fetch the arguments used in subscripting. 

As a sidenote - `__getitem__` is already part of `nodes.ClassDef` since it [inherits](https://github.com/pylint-dev/astroid/blob/cdeb259e4258f7472c526cf373c35c2600f1c3c6/astroid/nodes/scoped_nodes/scoped_nodes.py#L1823C43-L1823C43) that method from [LocalsDictNodeNG](https://github.com/pylint-dev/astroid/blob/cdeb259e4258f7472c526cf373c35c2600f1c3c6/astroid/nodes/scoped_nodes/mixin.py#L139C7-L139C7).

File: ChangeLog [hunk 2]
```diff
@@ -205,6 +201,10 @@ Release date: TBA
 
   Closes pylint-dev/pylint#8749
 
+* Fix a crash when inferring a `typing.TypeVar` call.
+
+  Closes pylint-dev/pylint#8802
+
```
Comment (jacobtylerwalls) [inline ChangeLog:207]: Actually 2.15.6 has since been released since my comment, so we need a new header for 2.15.7 (and to add the release date for 2.15.6)

Comment (jacobtylerwalls) [reply ChangeLog:207]: Also it takes double backticks for code literals.

File: astroid/brain/brain_typing.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (DanielNoord) [reply astroid/brain/brain_typing.py:137]: Okay! Can we then remove that code?

[the remark this answers is not in the document]
Comment (mbyrnepr2) [reply astroid/brain/brain_typing.py:137]: I took a look into that just now. The template is used by [this function](https://github.com/pylint-dev/astroid/blob/main/astroid/brain/brain_typing.py#L187) which needs the `__args__` member since the function is inferring subscriptable types.
As an illustration, If I were to remove the code and use a similar ClassDef construction for the code above, this unitest fails:
`FAILED astroid/tests/brain/test_brain.py::TypingBrain::test_has_dunder_args - astroid.exceptions.InferenceError: StopIteration raised without any error information.`
In other words, running Pylint on this code:
```python
from typing import Union
NumericTypes = Union[int, float]
NumericTypes.__args__ #@
```
...gives: `E1101: Class 'Union' has no '__args__' member (no-member)`

**Review by DanielNoord — APPROVED** [review]
